connect/runtime/src/test/java/org/apache/kafka/connect/integration/TransformationIntegrationTest.java

```diff
@@ -318,4 +323,61 @@ public void testFilterOnHasHeaderKeyWithSourceConnector() throws Exception {
         // delete connector
         connect.deleteConnector(CONNECTOR_NAME);
     }
+
+    /**
+     * Test the {@link Filter} transformer with a
+     * {@link HasHeaderKey} predicate on a source connector.
+     */
+    @Test
+    public void testFilterOnHasHeaderKeyWithSourceConnectorAndTopicCreation() throws Exception {
```

Comment:
Do we need another test here, or could we essentially replace the previous test with this one?

Either way, the test that uses topic creation should probably have a comment (either in the method JavaDoc or in a line comment below) that explicitly mentions enabling topic creation. We don't want someone removing the topic creation configs at a later time.

WDYT?

Reply:
Since we have a few other tests replacing should be fine. 

I'll add comments. But also, the test will fail if we don't create it (I removed the admin command) because we disable auto topic creation at the broker.

Reply:
Sounds good. I just want to avoid someone trying to simplify the tests in the future without understanding that this test is verifying both features work together.

Reply:
I added the logic to the previous test. 

Added another unit test to test 2 groups with 2 transformations.